test/unit/ngsiv2/provisioning/listProvisionedDevices-test.js

```diff
@@ -0,0 +1,461 @@
+                should.exist(parsedBody.attributes[0].entity_type);
+                should.exist(parsedBody.attributes[1].expression);
+                should.exist(parsedBody.attributes[2].reverse);
+                parsedBody.attributes[0].entity_name.should.equal('Higro2000');
+                parsedBody.attributes[0].entity_type.should.equal('Higrometer');
+                parsedBody.attributes[1].expression.should.equal('${@humidity * 20}');
+                parsedBody.attributes[2].reverse.length.should.equal(2);
+                done();
+            });
+        });
+    });
+    describe('When a request for an unexistent device arrives', function() {
+        var options = {
+            url: 'http://localhost:' + iotAgentConfig.server.port + '/iot/devices/Light84',
+            headers: {
+                'fiware-service': 'smartGondor',
+                'fiware-servicepath': '/gardens'
+            },
+            method: 'GET'
+        };
+
+        it('should return a 404 error', function(done) {
+            request(options, function(error, response, body) {
+                should.not.exist(error);
+                response.statusCode.should.equal(404);
+                done();
+            });
+        });
+    });
+
+    describe('When a request for listing all the devices with a limit of 3 arrives', function() {
+        var options = {
+            url: 'http://localhost:' + iotAgentConfig.server.port + '/iot/devices?limit=3',
+            headers: {
+                'fiware-service': 'smartGondor',
+                'fiware-servicepath': '/gardens'
+            },
+            method: 'GET'
+        };
+
+        function createDeviceRequest(i, callback) {
+            /* jshint camelcase: false */
+
+            var provisioningDeviceOptions = {
+                url: 'http://localhost:' + iotAgentConfig.server.port + '/iot/devices',
+                method: 'POST',
+                headers: {
+                    'fiware-service': 'smartGondor',
+                    'fiware-servicepath': '/gardens'
+                },
+                json: utils.readExampleFile('./test/unit/examples/deviceProvisioningRequests/provisionNewDevice.json')
+            };
+
+            provisioningDeviceOptions.json.devices[0].device_id =
+                provisioningDeviceOptions.json.devices[0].device_id + '_' + i;
+
+            request(provisioningDeviceOptions, callback);
+        }
+
+        beforeEach(function(done) {
+            nock.cleanAll();
+
+            contextBrokerMock = nock('http://192.168.1.1:1026')
+                .matchHeader('fiware-service', 'smartGondor')
+                .matchHeader('fiware-servicepath', '/gardens')
+                .post('/v2/registrations')
+                .times(10)
+                .reply(201, null, {'Location': '/v2/registrations/6319a7f5254b05844116584d'});
+
+            contextBrokerMock
+                .post('/v2/entities?options=upsert')
```

Comment:
Mock using post without payload. Comment explaining it needed?

Reply:
Fixed with fbc86fa76f4561b711a63053695efd21941cf4e5